graph/api/src/main/scala/org/apache/spark/graph/api/PropertyGraphType.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.graph.api
+
+/**
+ * Describes the structure of a [[PropertyGraph]].
+ *
+ * @since 3.0.0
+ */
+trait PropertyGraphType {
```

Comment:
Is ID types and property types part of a graph schema?

Reply:
No, it just contains meta data about the label sets and relationship types in the graph. We thought it would be redundant as you can do `graph.nodeFrame(Array("Foo", "Bar")).df.schema`.